### PR TITLE
Circuit optimization support for IdentityGate

### PR DESCRIPTION
**Pull request by mpharrigan**
- IdentityGate to support _phase_by_
 - DropIdentities optimization

### CODE REVIEW

**Comment by mpharrigan** [comment]
lemme add some tests

File: cirq/__init__.py [hunk 1]
```diff
@@ -197,6 +197,7 @@
 from cirq.optimizers import (
     ConvertToCzAndSingleGates,
     DropEmptyMoments,
+    DropIdentities,
```
Comment (Strilanc) [inline cirq/__init__.py]: I think this should just be part of `DropNegligible`. Give `IdentityGate` a `def _trace_distance_bound_(self): return 0` and it should work.

**Comment by dabacon** [comment]
Should we close this?  Identity now has trace distance 0.

**Comment by mpharrigan** [comment]
Yes

**Pull request closed without merging**